data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -211,4 +218,398 @@ public static InputStream resolveInputStream(HttpResponse response, InputStream
         }
     }
 
+    public static boolean isNullOrEmpty(String str) {
+        return str == null || CoreUtils.isNullOrEmpty(str.trim());
+    }
+
+    /**
+     * Checks if a CharSequence is empty (""), null or whitespace only.
+     * 
+     * Whitespace is defined by {@link Character#isWhitespace(char)}.
+     *
+     * @param cs the CharSequence to check, may be null
+     * @return true if the CharSequence is empty or null or whitespace only
+     */
+    public static boolean isBlank(CharSequence cs) {
+        if (cs == null) {
+            return true;
+        }
+        int strLen = cs.length();
+        if (strLen == 0) {
+            return true;
+        }
+        for (int i = 0; i < strLen; i++) {
+            if (!Character.isWhitespace(cs.charAt(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Checks if a CharSequence is not empty (""), not null and not whitespace only.
+     * 
+     * Whitespace is defined by {@link Character#isWhitespace(char)}.
+     *
+     * @param cs the CharSequence to check, may be null
+     * @return true if the CharSequence is not empty and not null and not whitespace only
+     */
+    public static boolean isNotBlank(CharSequence cs) {
+        return !isBlank(cs);
+    }
+
+    /**
+     * Removes the last character from a string.
+     * 
+     * If the string ends in {@code \r\n}, then remove both of them.
+     *
+     * @param str the string to chop
+     * @return the string with the last character removed, or null if null string input
+     */
+    public static String chop(final String str) {
```

Comment:
I don't like the special case.
Make it into "chop" and "chopNewLine" or something